tests/ci/docker_images/linux-x86/amazonlinux-2022_base/Dockerfile

```diff
@@ -0,0 +1,50 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0 OR ISC
+
+FROM amazonlinux:2022
+
+SHELL ["/bin/bash", "-c"]
+
+ENV DEPENDENCIES_DIR=/home/dependencies
+ENV LLVM_PROJECT_HOME=${DEPENDENCIES_DIR}/llvm-project
+# Avoid "Use -buildvcs=false to disable VCS stamping." errors due to newer versions of Go
+ENV GOFLAGS="-buildvcs=false"
+
+# llvm, llvm-devel, libcxx, and libcxxabi are needed for the sanitizer tests.
+# llvmorg-15.0.6 is the latest stable release as of 2023-01-05.
+# See https://github.com/google/sanitizers/wiki/MemorySanitizerLibcxxHowTo
+# Install Valgrind for Valgrind test target even though it is not needed for the base test target.
+RUN set -ex && \
+    yum -y update && yum install -y \
+    cmake3 \
+    ninja-build \
+    perl \
+    golang \
+    which \
+    git \
+    ca-certificates \
+    wget \
+    llvm \
+    llvm-devel \
+    # valgrind is just the application and core resources.
+    valgrind \
+    # valgrind/memcheck.h is provided by the valgrind-devel package on AL2. see P63119011.
+    valgrind-devel \
+    unzip && \
+    # Based on https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2-linux.html
+    curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip" && \
+    unzip awscliv2.zip && \
+    ./aws/install --bin-dir /usr/bin && \
+    rm -rf awscliv2.zip aws/ && \
+    mkdir -p ${DEPENDENCIES_DIR} && \
+    cd ${DEPENDENCIES_DIR} && \
+    git clone https://github.com/llvm/llvm-project.git --branch llvmorg-15.0.6  --depth 1 && \
```

Comment:
Why does this require llvm? Also why version 15 when the al2022 version of clang is 14?

Reply:
To run the memory sanitizer without false positives from inside the C standard functions we need to compile libc as a [part of our build](https://github.com/awslabs/aws-lc/blob/main/CMakeLists.txt#L705-L772). I didn't think about the clang 14 vs llvm 15. This only needs the libcxx and libcxxabi parts, but it makes more sense to use the matching set.

Reply:
Turns out nothing was using this and trying to add the sanitizer results in C++11 vs 17 version issues. I'll tackle this all later.